src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -121,6 +123,13 @@ public string SelectedLanguage
             }
             set
             {
+                if (selectedLanguage != value && selectedLanguage != null)
+                {
+                    Dynamo.Logging.Analytics.TrackEvent(
+                        Actions.Switch,
+                        Categories.Preferences,
+                        Res.PreferencesViewLanguageLabel);
```

Comment:
This pattern introduce some repeated code but will be most accurate to avoid logging dialog initialization